src/alignment/graphTest/Makefile

```diff
@@ -1,8 +1,20 @@
-main: main.o Graph.o
-	g++ main.o Graph.o -o main
-main.o: main.cpp Graph.h
-	g++ -c main.cpp
-Graph.o: Graph.cpp Graph.h
-	g++ -c Graph.cpp
-clean:
-	rm *.o main
+CC = g++
+OBJ = main.o Graph.o Node.o
+HEADER1 = ../Graph.h
+HEADER2 = ../Node.h 
+CFLAGS = -c -Wall -I../
+all: main
+main: $(OBJ) 
+	$(CC) $(OBJ) -o $@
+
+main.o: main.cpp $(HEADER1)

```

Comment:
Don't call the program `main`. It is too unspecific and will trigger name clashes if this is the general policy.